src/modules/Auth/services/auth.ts

```diff
@@ -1,26 +1,81 @@
-export interface User {
-  name: string;
-  email: string;
-  password?: string;
+import Keycloak, { KeycloakConfig } from 'keycloak-js';
+
+const onLoad: Keycloak.KeycloakOnLoad = 'login-required';
+
+class Config implements KeycloakConfig {
+  url: string = process.env.AUTH_URL || 'auth';
+  realm: string = process.env.AUTH_REALM || 'sume';
+  clientId: string = process.env.AUTH_CLIENT_ID || 'sume-app';
 }
 
-export function SignIn(): Promise<any> {
-  return new Promise((resolve) =>
-    setTimeout(
-      () =>
-        resolve({
-          token: 'fake-token',
-          user: { name: 'Ricardo Lüders', email: 'my@email.com' },
-        }),
-      1000,
-    ),
-  );
+const _kc = Keycloak(new Config());
+
+function Init(appEntrypoint: () => void): void {
+  _kc
+    .init({
+      onLoad,
+    })
+    .then((auth) => {
+      if (!auth) {
+        window.location.reload();
+      } else {
+        _kc
+          .loadUserProfile()
+          .then(() => {
+            appEntrypoint();
+          })
+          .catch((err) => {
+            console.error('Cannot get user profile: ', err);
```

Comment:
we should think how handle if user has no profile.
Maybe redirecting to profile page or displaying some message asking to complete the profile.
But we can open a new issue to handle with this

Reply:
Yes, I'm still not sure about Keycloak's capabilities on this subject. Maybe it can handle this sort of scenario. If it does, we can use it. I can check.